default.nix

```diff
@@ -1,157 +1,92 @@
-      "registry+https://github.com/rust-lang/crates.io-index".curl-sys."*" = with pkgs; [ nghttp2 ] ++ macosFrameworks;
-      "registry+https://github.com/rust-lang/crates.io-index".libgit2-sys."*" = with pkgs; [ libiconv ] ++ macosFrameworks;
-    };
-  };
-
-  rustPackages = pkgs.callPackage ./crate.nix {
-    inherit packageFun rustc cargo resolver;
-    config = config pkgs;
-    buildConfig = config pkgs.buildPackages;
-  };
-
-  # done
-
-  # your rust build is available here
-  package = rustPackages.unknown.cargo2nix.${version} {
-    freezeFeatures = true;
-    meta.platforms = lib.platforms.darwin ++ lib.platforms.linux;
-  };
-
-  # how to use cargo2nix to speed up resolution:
-
-  resolveResponse =
+  packageFun = import ./Cargo.nix { };
+  rustPackageConfig =
     let
-      request =
-        builtins.toFile
-          "resolve-request.json"
-          (builtins.toJSON
-            (pkgs.rustBuilder.rustLib.buildResolveRequest {
-              initial = [
-                {
-                  package-id = "cargo2nix ${version}";
-                }
-              ];
-              inherit (pkgs) stdenv;
-              inherit packageFun;
-            }));
+      darwinFrameworks = lib.optionals pkgs.hostPlatform.isDarwin
+        (with pkgs.darwin.apple_sdk.frameworks; [ Security CoreServices ]);
     in
-    lib.importJSON
-      (pkgs.runCommand
-        "resolve"
-        { nativeBuildInputs = [package]; }
-        "cargo2nix resolve <${request} >$out");
-
-  rustPackagesWithResolve = pkgs.callPackage ./crate.nix {
-    inherit packageFun rustc cargo resolver;
-    config = config pkgs // { resolve = resolveResponse; };
-    buildConfig = config pkgs.buildPackages // { resolve = resolveResponse; };
-  };
-
-in
-{
-  inherit package;
-
-  package' = rustPackagesWithResolve.unknown.cargo2nix.${version} {};
-
-
-  # and you can make a development shell
-  shell = pkgs.rustBuilder.makeShell {
-    inherit packageFun cargo rustc;
-    packageResolver = { source, name, version, sha256, ... }:
       {
-        src = resolver { inherit source name version; };
+        rustcflags = {
+          "registry+https://github.com/rust-lang/crates.io-index"."*" = [ "--cap-lints" "warn" ];
+        };
+        nativeBuildInputs = {
+          "registry+https://github.com/rust-lang/crates.io-index".curl-sys."*" = darwinFrameworks;
+          "registry+https://github.com/rust-lang/crates.io-index".libgit2-sys."*" = darwinFrameworks;
+        };
+        buildInputs = {
+          "registry+https://github.com/rust-lang/crates.io-index".libgit2-sys."*" = [ pkgs.libiconv ];
+          "registry+https://github.com/rust-lang/crates.io-index".cargo."*" = [ pkgs.libiconv ] ++ darwinFrameworks;
+          unknown.cargo2nix2."*" = [ pkgs.libiconv ] ++ darwinFrameworks;
+        };
+        environment = {
+          "registry+https://github.com/rust-lang/crates.io-index".openssl-sys."*" =
```

Comment:
This entire let-in block seems quite complex to include in the `default.nix`. Could this maybe be added to the overlay and re-exported so it can be re-used, as suggested in #19?

Reply:
Good idea! I'll leave this to a later PR where I plan to introduce a more composable way to provide config.

Reply:
Sounds good! 👌